src/html-imports.js

```diff
@@ -9,718 +9,754 @@
+      const o = s$[i];
+      const assetpath = o.getAttribute('assetpath') || '';
+      o.setAttribute('assetpath', Path.replaceAttrUrl(assetpath, url));
     }
   }
-  fixUrlsInTemplates(content, base);
-}
 
-function fixUrlsInTemplates(element, base) {
-  var t$ = element.querySelectorAll('template');
-  for (var i=0; i < t$.length; i++) {
-    fixUrlsInTemplate(t$[i], base);
+  /**
+   * Replaces all the imported scripts with a clone in order to execute them.
+   * Updates the `currentScript`.
+   * @param {!HTMLDocument} doc
+   * @return {Promise} Resolved when scripts are loaded.
+   */
+  function runScripts(doc) {
+    const s$ = doc.querySelectorAll(`script[type=${scriptType}]`);
+    let promise = Promise.resolve();
+    for (let i = 0, l = s$.length, s; i < l && (s = s$[i]); i++) {
+      promise = promise.then(() => {
+        const c = doc.createElement('script');
+        c.textContent = s.textContent;
+        if (s.src) {
+          c.setAttribute('src', s.getAttribute('src'));
+        }
+        // Listen for load/error events before adding the clone to the document.
+        // Catch failures, always return c.
+        const whenLoadedPromise = whenElementLoaded(c).catch(() => c);
+        // Update currentScript and replace original with clone script.
+        currentScript = c;
+        s.parentNode.replaceChild(c, s);
+        // After is loaded, reset currentScript.
+        return whenLoadedPromise.then((script) => {
+          if (script === currentScript) {
+            currentScript = null;
+          }
+        });
+      });
+    }
+    return promise;
   }
-}
 
-function fixUrls(element, base) {
-  var n$ = element.querySelectorAll(importsSelectors);
-  for (var i=0; i < n$.length; i++) {
-    var n = n$[i];
-    if (n.href) {
-      n.href = new URL(n.getAttribute('href'), base);
-    }
-    if (n.src) {
-      n.src = new URL(n.getAttribute('src'), base);
-    }
-    if (n.localName == 'style') {
-      path.resolveUrlsInStyle(n, base);
+  /**
+   * Waits for all the imported stylesheets/styles to be loaded.
+   * @param {!HTMLDocument} doc
+   * @return {Promise}
+   */
+  function waitForStyles(doc) {
+    const s$ = doc.querySelectorAll(stylesInImportsSelector);
+    const promises = [];
+    for (let i = 0, l = s$.length, s; i < l && (s = s$[i]); i++) {
+      // Catch failures, always return s
+      promises.push(
+        whenElementLoaded(s).catch(() => s)
+      );
     }
+    return Promise.all(promises);
   }
-  fixUrlsInTemplates(element, base);
-}
 
-var scriptType = "import-pending";
+  /**
+   * Clones styles and stylesheets links contained in imports and moves them
+   * as siblings of the root import link.
+   * @param {!HTMLLinkElement} importLink
+   */
+  function cloneAndMoveStyles(importLink) {
+    const n$ = importLink.import.querySelectorAll(stylesSelector);
+    for (let i = 0, l = n$.length, n; i < l && (n = n$[i]); i++) {
+      const clone = document.createElement(n.localName);
```

Comment:
Clones of links that already had a parent won't work, will document https://gist.github.com/valdrinkoshi/4a92f97169a6fc41a1852f23211b8c4e